src/shared/util/validation.ts

```diff
@@ -0,0 +1,60 @@
+import validator from 'validator'
+import { parse } from 'url'
+
+import blacklist from '../../server/resources/blacklist'
+
+export const WHITELIST = [new RegExp('^http://localhost:4572')]
+
+export const URL_OPTS: ValidatorJS.IsURLOptions = {
+  protocols: ['https'],
+  require_tld: true,
+  require_protocol: true,
+  require_host: true,
+  require_valid_protocol: true,
+  allow_underscores: false,
+  allow_trailing_dot: false,
+  allow_protocol_relative_urls: false,
+  disallow_auth: false,
+}
+
+// Checks if url is whitelisted.
+export function isWhitelisted(url: string): boolean {
+  return WHITELIST.some((regexp) => url.match(regexp))
+}
+
+// Checks if url is blacklisted.
+export function isBlacklisted(url: string): boolean {
+  return blacklist.some((bl) => url.includes(bl))
+}
+
+// Tests if a URL string begins with https://.
+export function isHttps(url: string, useWhitelist = false): boolean {
+  if (useWhitelist && isWhitelisted(url)) return true
+  return /^https:\/\//.test(url)
+}
+
+// Check if a URL string is valid.
+export function isValidUrl(url: string, useWhitelist = false): boolean {
+  if (useWhitelist && isWhitelisted(url)) return true
+  return validator.isURL(url, URL_OPTS)
```

Comment:
does this _guarantee_ that there is no change in behavior from the previous implementation?

Reply:
The only change in this function is the addition of line 38, which will only be activated if `useWhitelist` is set as true. This change is backwards compatible as the new parameter is optional. And hence, there should not be a change of behaviour.